App/Sagas/PhotoViewingSagas.ts

```diff
@@ -1,20 +1,37 @@
-import { call, put, select } from 'redux-saga/effects'
-import { ActionType } from 'typesafe-actions'
+import { delay } from 'redux-saga'
+import { call, put, select, take } from 'redux-saga/effects'
+import { ActionType, getType } from 'typesafe-actions'
 
 import PhotoViewingActions, { ThreadData } from '../Redux/PhotoViewingRedux'
-import { photoAndComment } from '../Redux/PhotoViewingSelectors'
-import UIActions from '../Redux/UIRedux'
+import { photoAndComment, shouldNavigateToNewThread, photoToShareToNewThread } from '../Redux/PhotoViewingSelectors'
 import TextileNode from '../../TextileNode'
-import { Threads, Thread, Photo, BlockId } from '../Models/TextileTypes'
+import { Threads, Photo, BlockId, PhotoId } from '../Models/TextileTypes'
 import NavigationService from '../Services/NavigationService'
+import { shareWalletImage } from './ImageSharingSagas'
+
+export function * monitorNewThreadActions () {
```

Comment:
This one listens for `threadAdded` actions and check if there are any pending requests to navigate to or share a photo to the new thread.